src/ion/liveranges.rs

```diff
@@ -1159,66 +1180,108 @@ impl<'a, F: Function> Env<'a, F> {
                     VRegIndex::new(vreg),
                     range,
                 );
-                let mut last_point = None;
-                let mut fixup_multi_fixed_vregs = |pos: ProgPoint,
-                                                   slot: usize,
-                                                   op: &mut Operand,
-                                                   fixups: &mut Vec<(
-                    ProgPoint,
-                    PRegIndex,
-                    PRegIndex,
-                    usize,
-                )>| {
-                    if last_point.is_some() && Some(pos) != last_point {
-                        seen_fixed_for_vreg.clear();
-                        first_preg.clear();
+
+                // Find groups of uses that occur in at the same program point.
+                for uses in self.ranges[range.index()]
+                    .uses
+                    .group_by_mut_(|a, b| a.pos == b.pos)
```

Comment:
This is a nice API but it's unfortunate that it's not yet stable, and that we have to copy part of stdlib into `util.rs` to get it; I'd prefer not to duplicate like that, and to add a few hundred lines of code to save just a few here. Until we have stabilized `group_by_mut`, could we just do the scan manually here?

Reply:
I actually tried to do the scan manually at first and the result was an unreadable mess. Hiding the logic to group entries by `ProgPoint` makes the loop *much* more readable.

I can shrink down the code in `util.rs` to only contain the parts that we actually needs (e.g. we don't use `group_by`, only `group_by_mut`). Alternatively, the itertools crate also provides a version of `group_by` (with slightly different arguments).

Reply:
Ah, I think using `itertools` is reasonable for now, then; it's a pretty small dep and if it improves readability here, then all the better. (IIRC there are similar scan-and-find-groups patterns in e.g. move resolution; maybe we can clean those up too later, but that's obviously outside the scope of this PR!)